src/main/java/loqor/ait/tardis/data/SiegeHandler.java

```diff
@@ -127,60 +125,52 @@ public void setActive(boolean siege) {
             TardisDesktop.playSoundAtConsole(console, sound, SoundCategory.BLOCKS, 3f, 1f);
         }
 
-        this.tardis.removeFuel((0.01 * FuelHandler.TARDIS_MAX_FUEL) * (this.tardis.travel().getHammerUses() + 1));
-
+        this.tardis.removeFuel(0.01 * FuelHandler.TARDIS_MAX_FUEL * this.tardis.travel().instability());
         this.active.set(siege);
     }
 
-    private static boolean hasLeatherArmour(ServerPlayerEntity player) {
-        int count = 0;
+    @Override
+    public void tick(MinecraftServer server) {
+        if (!this.active.get())
+            return;
 
-        for (ItemStack item : player.getArmorItems()) {
-            if (item.isOf(Items.LEATHER_HELMET) || item.isOf(Items.LEATHER_CHESTPLATE)
-                    || item.isOf(Items.LEATHER_LEGGINGS) || item.isOf(Items.LEATHER_BOOTS)) {
-                count++;
-            }
-        }
+        this.siegeTime.flatMap(i -> this.active.get() ? i + 1 : 0);
 
-        return count == 4;
-    }
+        if (server.getTicks() % 20 != 0)
+            return;
 
-    @Override
-    public void tick(MinecraftServer server) {
-        if (this.isSiegeBeingHeld() && tardis().getExterior().findExteriorBlock().isPresent())
+        boolean isHeld = this.isSiegeBeingHeld();
+
+        if (isHeld && this.tardis.getExterior().findExteriorBlock().isPresent())
             this.setSiegeBeingHeld(null);
 
-        int siege_time = this.getTimeInSiegeMode() + 1;
-        this.siegeTime.set(this.active.get() ? siege_time : 0);
+        boolean freeze = !isHeld && this.getTimeInSiegeMode() > 60 * 20;
 
-        if (!this.active.get())
-            return;
+        for (ServerPlayerEntity player : TardisUtil.getPlayersInsideInterior(this.tardis.asServer())) {
+            if (!player.isAlive())
+                continue;
 
-        // todo add more downsides the longer you are in siege mode as it is meant to
-        // fail systems
-        // and
-        // kill you and that
-        if (this.getTimeInSiegeMode() > (60 * 20) && !this.isSiegeBeingHeld()) {
-            for (ServerPlayerEntity player : TardisUtil.getPlayersInsideInterior(this.tardis)) {
-                if (!player.isAlive())
-                    continue;
-                if (hasLeatherArmour(player)) {
```

Comment:
leather armour resistance seems to have been lost

Reply:
Not true
The check was moved and replaced with a vanilla one

Reply:
okie dokie